components/LoginForm/LoginForm.jsx

```diff
@@ -1,23 +1,35 @@
-import { useState } from "react"
+import { useState, useReducer } from "react"
 import { useUser } from "@supabase/auth-helpers-react"
 import { createClientComponentClient } from "@supabase/auth-helpers-nextjs"
 import Link from "next/link"
 import styles from "./LoginForm.module.css"
 import { useRouter } from "next/router"
+import InputReducer from "../InputReducer/InputReducer"
+import Loader from "@/components/Loader/Loader"
 
 export default function LoginForm() {
   const supabase = createClientComponentClient()
   const user = useUser()
   const router = useRouter()
-  const [loading, setLoading] = useState(true)
-  const [email, setEmail] = useState("")
-  const [password, setPassword] = useState("")
+  // const [loading, setLoading] = useState(true)
+  const [formValue, dispatch] = useReducer(InputReducer, {
+    email: "",
+    password: "",
+    submitting: false,
+    success: false,
+    error: null,
+  })
```

Comment:
I find it more readable if you set the initial value to a variable:

```jsx
const initialState = {
  email: "",
  password: "",
  submitting: false,
  success: false,
  error: null,
}

export default function LoginForm() {
  const [state, dispatch] = useReducer(inputReducer, initialState)
}
```

Use `state` and `dispatch` so they match [documented conventions](https://react.dev/reference/react/useReducer)

Reply:
I think the ui.dev [multistep form reducer challenge](https://ui.dev/c/react/multistep-form-reducer-challenge) solution is a good example to compare your work to.